sql/mdl.cc

```diff
@@ -242,6 +277,21 @@ class Deadlock_detection_visitor: public MDL_wait_for_graph_visitor
           deadlocks are even shorter typically.
   */
   static const uint MAX_SEARCH_DEPTH= 32;
+
+  /* struct of store mdl deadlock information*/
+  struct MDL_deadlock_info {
+    char *thread_info;
```

Comment:
I do not think you need length anywhere, you can just store null terminated `char *` string.or you can store std::string.